### DISCUSSION TITLE
Ignore properties and methods with angular.

### DISCUSSION THREAD

**Discussion by SebastianStehle** [storybookjs/storybook#17910] [General]
Hi,

I have used storybook several times with react and would like to integrate it to an angular app now. I am confused why storybook creates controls for "internal" properties and methods. For example I have a component like this:

```
class DateTimeEditorComponent {
    public readonly timeControl = new FormControl();
    public readonly dateControl = new FormControl();
}
```

I see both properties in my story and storybook even overwrites them with undefined.

My story is not very special:

```
export default {
    title: 'Framework/DateTimeEditor',
    component: DateTimeEditorComponent
} as Meta;

const Template: Story<DateTimeEditorComponent> = (args: DateTimeEditorComponent) => ({
    props: args,
});

export const Date = Template.bind({});
```

**Comment by peckomingo** [comment]
We have the same problem. Controls panel is cluttered with unrelated stuff.

**Comment by fischerman** [comment]
Related to https://github.com/storybookjs/storybook/pull/18880
